lib/ngx/process.lua

```diff
@@ -0,0 +1,56 @@
+-- Copyright (C) Yichun Zhang (agentzh)
+
+
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+local getfenv = getfenv
+local errmsg = base.get_errmsg_ptr()
+local FFI_ERROR = base.FFI_ERROR
+local ffi_str = ffi.string
+
+local process_types = {
+    [base.FFI_PROCESS_SINGLE    ]  = "single",
+    [base.FFI_PROCESS_MASTER    ]  = "master",
+    [base.FFI_PROCESS_SIGNALLER ]  = "signaller",
+    [base.FFI_PROCESS_WORKER    ]  = "worker",
+    [base.FFI_PROCESS_HELPER    ]  = "helper",
+    [base.FFI_PROCESS_PRIVILEGED]  = "privileged agent",
+}
+
+
+local C = ffi.C
+local _M = { version = base.version }
+
+
+ffi.cdef[[
+int ngx_http_lua_ffi_enable_privileged_agent(char **err);
+int ngx_http_lua_ffi_get_process_type(void);
+]]
+
+
+function _M.type(is_str_name)
+    local typ = C.ngx_http_lua_ffi_get_process_type()
+    if is_str_name then
```

Comment:
We should always return the string names. Note that Lua's strings are always intern'd, so we can use it as "symbols", just like named constants. String comparisons in Lua is always `O(1)` operation, just like number comparisons.